### ISSUE TITLE
Sampled data is not able to be cached

### ISSUE THREAD

**Issue by edgararuiz** [rstudio/sparklyr#917]
Issue reported by one of our customer (Ticket 21412):

```
new_flight <- new_flight %>% compute("cached_flight")
new_sample <- new_flight %>% sample_n(10) %>% compute("my_sample")
Error: org.apache.spark.sql.AnalysisException: cannot recognize input near '(' 'cached_flight' 'TABLESAMPLE' in from source; line 3 pos 6
	at org.apache.spark.sql.hive.HiveQl$.createPlan(HiveQl.scala:318)
	at org.apache.spark.sql.hive.ExtendedHiveQlParser$$anonfun$hiveQl$1.apply(ExtendedHiveQlParser.scala:41)
	at org.apache.spark.sql.hive.ExtendedHiveQlParser$$anonfun$hiveQl$1.apply(ExtendedHiveQlParser.scala:40)
	at scala.util.parsing.combinator.Parsers$Success.map(Parsers.scala:136)
```

**Comment by javierluraschi** [comment]
```r
library(sparklyr)
sc <- spark_connect(master = "local")

library(dplyr)
copy_to(sc, iris, overwrite = TRUE)
dbGetQuery(sc, "SELECT * FROM (iris TABLESAMPLE (10 rows))")
```

```
   Sepal_Length Sepal_Width Petal_Length Petal_Width Species
1           5.1         3.5          1.4         0.2  setosa
2           4.9         3.0          1.4         0.2  setosa
3           4.7         3.2          1.3         0.2  setosa
4           4.6         3.1          1.5         0.2  setosa
```
but

```r
library(sparklyr)
sc <- spark_connect(master = "local", version = "1.6.1")

library(dplyr)
copy_to(sc, iris, overwrite = TRUE)

dbGetQuery(sc, "SELECT * FROM (iris TABLESAMPLE (10 rows))")
```

```
 Error: org.apache.spark.sql.AnalysisException: cannot recognize input near '(' 'iris' 'TABLESAMPLE' in from source; line 1 pos 15
	at org.apache.spark.sql.hive.HiveQl$.createPlan(HiveQl.scala:318)
	at org.apache.spark.sql.hive.ExtendedHiveQlParser$$anonfun$hiveQl$1.apply(ExtendedHiveQlParser.scala:41)
	at org.apache.spark.sql.hive.ExtendedHiveQlParser$$anonfun$hiveQl$1.apply(ExtendedHiveQlParser.scala:40)
	at scala.util.parsing.combinator.Parsers$Success.map(Parsers.scala:136)
	at scala.util.parsing.combinator.Parsers$Success.map(Parsers.scala:135)
```

Looks like the slow workaround was implemented as a stop-gap for `1.6` lacking sample functionality.

**Comment by javierluraschi** [comment]
We improved the error message, closing.
